pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -107,6 +113,10 @@ type BuildSpec struct {
 	// +optional
 	// +kubebuilder:validation:Format=duration
 	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// Env contains additional environment variables that should be passed to the build container
+	// +optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
```

Comment:
I know the SHIP called for using k8s corev1.EnvVar.  

Went back and revisited https://github.com/shipwright-io/community/pull/10 where te SHIP was introduced.  Opened up all the unresolved conversations.  Saw no mention of what I'm about to say there.

Do we in fact intend to support all the k8s env var options as spelled out at https://github.com/kubernetes/api/blob/7036ead253974a37c7aca010f305593a486a253e/core/v1/types.go#L1931-L2055 ?

There is *A LOT* of flexibility and permutation there.

And yes, at a high level, since we are mapping things to the container env var fields, there is a "it will just work" mentality which seems OK.

But have we worked through all those different sources for EnvVars and are sure they will work seamlessly?

Because this is often not the case with openshift, and we've become much more selective when choosing to use k8s core types, and often create subset API for stuff like this.

@coreydaley - I think all of ^^ is a question for @adambkaplan as SHIP author, with the admission up front that I reviewed that SHIP and this did not occur to me until now (perhaps because I had not yet received the openshift API education I just referenced).

But before we go too much further, WDYT @adambkaplan ?  Am I too paranoid here?  Or based on recent experiences elsewhere, do we want to step back a bit, and be more conservative on env vars source options we introduced?  At least initially ...